### PR TITLE
$links as private variable

### PR DESCRIPTION
**Pull request by totoprayogo1916**
**Description**
Convert `$links` as private variable

**Checklist:**
- [x] Securely signed commits
- [ ] Component(s) with PHPdocs
- [ ] Unit testing, with >80% coverage
- [ ] User guide updated
- [ ] Conforms to style guide

### CODE REVIEW

File: system/Pager/PagerRenderer.php [hunk 1]
```diff
@@ -282,20 +289,18 @@ public function getCurrent(): string
 	 */
 	public function links(): array
 	{
-		$links = [];
-
```
Comment (samsonasik) [inline system/Pager/PagerRenderer.php:286]: if it going to happen, it needs return early like this:

if $this->links !== [] 
{
    return $this->links; 
}

**Comment by samsonasik** [comment]
could you provide the use case for make it property?

File: system/Pager/PagerRenderer.php [hunk 2]
```diff
@@ -76,6 +76,13 @@ class PagerRenderer
 	 */
 	protected $pageSelector;
 
+	/**
+	 * Links to dipsplay
```
Comment (MGatner) [inline system/Pager/PagerRenderer.php:80]: Typo
```suggestion
	 * Links to display
```

File: system/Pager/PagerRenderer.php [hunk 3]
```diff
@@ -76,6 +76,13 @@ class PagerRenderer
...
+	 *
+	 * @var array
+	 */
+	private $links = [];
```
Comment (MGatner) [inline system/Pager/PagerRenderer.php:84]: `$last` technically can be `< 1` so one empty array is a valid return for `links()` already. This should start as `null` and be checked as such for processing.
```suggestion
	 * @var array|null
	 */
	private $links;
```

File: system/Pager/PagerRenderer.php [hunk 4]
```diff
@@ -282,20 +289,23 @@ public function getCurrent(): string
 	 */
 	public function links(): array
 	{
-		$links = [];
+		if ($this->links !== [])
+		{
+			return $this->links;
+		}
```
Comment (MGatner) [inline system/Pager/PagerRenderer.php:296]:
```suggestion
		if (isset($this->links))
		{
			return $this->links;
		}

		$this->links = [];
```

**Review by MGatner — CHANGES_REQUESTED** [review]

**Comment by MGatner** [comment]
@totoprayogo1916 did you abandon this?

**Comment by totoprayogo1916** [comment]
> @totoprayogo1916 did you abandon this?

I think yes @MGatner . It might even be complicated.

**Pull request closed without merging**